src/ddx_gradients.f90

```diff
@@ -1603,11 +1623,592 @@ subroutine contract_grad_f_worker2(params, constants, workspace, &
 
     deallocate(gradpsi_grid, stat=istat)
     if (istat.ne.0) then
-        workspace % error_flag = 1
-        workspace % error_message = "deallocation error in ddx_grad_f_worker2"
+        call update_error(error, "deallocation error in ddx_grad_f_worker2")
         return
     end if
 
 end subroutine contract_grad_f_worker2
 
+!> Sphere contribution to the ddPCM matrix gradient using N^2 code
+subroutine gradr_sph(params, constants, isph, vplm, vcos, vsin, basloc, &
```

Comment:
Did you do this on purpose? Seems to have nothing to do with error rework.

Reply:
I moved it here from the file `ddx_pcm.f90`, as it make more sense for it to be in `ddx_gradients.f90`.
Indeed, it is not related to the error rework, but it is a more general refactor.

I plan to move it there in any case, should I restore it and move it in the next pull request?

Reply:
if it's just moving code, I think that's fine.